### PR TITLE
Implement onboarding flow & integrate user profiles with server

### PR DESCRIPTION
**Pull request by andrewbailey**
Onboarding is now implemented and the profile page now accesses the server. Also, the app now properly handles configuration changes.

Edge cases & configuration changes handled (please test these if possible):
 - Logging into an account, and restarting the app without completing onboarding still takes the user to the onboarding page
 - Rotating the screen when editing the profile does not lose any data
 - Rotating the screen when saving the profile keeps the new activity in the same saving state

### CODE REVIEW

File: app/src/main/java/com/mealsmadeeasy/ui/home/profile/ProfileFragment.kt [hunk 1]
```diff
@@ -94,11 +134,84 @@ class ProfileFragment : BaseFragment() {
...
 
         return root
     }
 
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+
+        outState.putInt(KEY_SAVED_SEX, sexSpinner.selectedItemPosition)
+        outState.putLong(KEY_SAVED_BDAY, cal.timeInMillis)
+        outState.putInt(KEY_SAVED_HEIGHT_FT, feetText.text.toString().toInt())
```
Comment (adurso3) [inline app/src/main/java/com/mealsmadeeasy/ui/home/profile/ProfileFragment.kt]: Not checking whether the text is actually convertible to an int will cause the app to crash when this method gets called on the initial onboarding screen.

File: app/src/main/res/layout/view_user_profile.xml [hunk 2]
```diff
@@ -1,95 +1,150 @@
...
+                    android:textColor="@android:color/black"
+                    android:text="@string/onboarding_header"/>
 
-            <android.support.design.widget.TextInputEditText
-                android:id="@+id/height_feet_edit_text"
+                <TextView
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    style="@style/TextAppearance.AppCompat.Body1"
+                    android:textColor="#616161"
```
Comment (adurso3) [inline app/src/main/res/layout/view_user_profile.xml]: Should also probably be a resource.

File: app/src/main/res/layout/view_user_profile.xml [hunk 3]
```diff
@@ -1,95 +1,150 @@
...
-            android:layout_weight="1"
-            android:layout_height="wrap_content"
-            android:hint="@string/profile_height_prompt" >
+            <LinearLayout
+                android:id="@+id/profile_onboarding_header"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:orientation="vertical"
+                android:paddingStart="16dp"
+                android:paddingEnd="16dp"
```
Comment (adurso3) [inline app/src/main/res/layout/view_user_profile.xml]: Resources please. ~~Revenge is sweet.~~

**Review by adurso3 — CHANGES_REQUESTED** [review]
Looks pretty good. I did check the edge cases you listed, and those are all behaving as expected.

**Comment by andrewbailey** [comment]
Should be resolved now.

**Review by adurso3 — APPROVED** [review]
